addons/payment_worldline/models/payment_transaction.py

```diff
@@ -154,17 +154,9 @@ def _worldline_create_checkout_session(self):
                 worldline_code = const.PAYMENT_METHODS_MAPPING.get(self.payment_method_id.code, 0)
                 payload['cardPaymentMethodSpecificInput']['paymentProductId'] = worldline_code
             else:
-                pm_codes = self.env['payment.method'].search([
-                    ('active', 'in', [True, False]),
-                    ('primary_payment_method_id', '=', self.payment_method_id.id),
-                ]).mapped('code')
-                worldline_codes = [
-                    const.PAYMENT_METHODS_MAPPING[code] for code in pm_codes
-                    if code in const.PAYMENT_METHODS_MAPPING
-                ]
```

Comment:
What if `self.payment_method_id.brand_ids` is True and it's not the card payment method?